### PR TITLE
packages: Add Dockerfile and documentation for local builds

### LINKED ISSUE #156: Allow local builds
Local packages can be created using the osc tool. This allow faster build times when testing new changes to the packages. An openSUSE-based (better osc support, plugins, etc) container can be used to perform local builds.

### PR DESCRIPTION
**Pull request by erick0z**
This commit adds a Dockerfile and proper documentation explaining how
to test changes with locally built packages using an openSUSE-based
container.

Fixes #156

Signed-off-by: Erick Cardona <erick.cardona.ruiz@intel.com>

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -78,3 +78,39 @@ Repositories
 .. _`clear-containers-image`: https://download.clearlinux.org/current/
 
 .. _`clear-containers-selinux`: https://github.com/clearcontainers/proxy/tree/master/selinux
+
+Local Building
+==============
+
+Any user can build their own Clear Containers packages using the source files provided in this
+repository, either using the OBS platform or in their own machines using the osc (Open Build Service command line tool).
```
Comment (jodh-intel) [inline README.rst]: s/using the osc/using the `osc`/.

File: README.rst [hunk 2]
```diff
@@ -78,3 +78,39 @@ Repositories
...
+
+A Dockerfile is provided to build a container that has all the tools needed to perform local builds.
+
+.. code-block:: shell
+
+   # cd into this repository
+   cd < this repo location>
```
Comment (jodh-intel) [inline README.rst]: Can you change all the commands to have a `$ ` prefix? Admittedly, this means it's slightly awkward to copy-and-paste, but that's the convention we always use to show commands the user runs.

File: README.rst [hunk 3]
```diff
@@ -78,3 +78,39 @@ Repositories
...
+
+   # build the container image
+   # cc-osc name is used for the tag, it can be anything.
+   docker build -t cc-osc .
```
Comment (jodh-intel) [inline README.rst]: The `docker build` commands should probably be prepended with `sudo` for maximum portability.

File: README.rst [hunk 4]
```diff
@@ -78,3 +78,39 @@ Repositories
...
+
+   # If you are behind a proxy
+   docker build --build-arg=http_proxy=http://<host>:<port> --build-arg=https_proxy=http://<host>:<port> -t cc-osc .
+
+   # Run the container and bind this repository
+   docker run -v /repo/location:/root/packaging -it cc-osc bash
```
Comment (jodh-intel) [inline README.rst]: Shouldn't this be the following?:

```
$ sudo docker run -v $PWD:/root/packaging -it cc-osc bash
```

File: README.rst [hunk 5]
```diff
@@ -78,3 +78,39 @@ Repositories
...
+
+   # To see the options, arguments, etc
+   ./update_runtime.sh --help
+
+   # to build locally
+   ./update_runtime.sh --local-build --verbose
+
+   # the resulting packages can be found in /var/packaging/results inside the container.
+   # It could be convenient to bind the /var/packaging/results directory to the host:
+   docker run -v /repo/location:/root/packaging -v /var/packages:/var/packaging/results -it cc-osc bash
```
Comment (jodh-intel) [inline README.rst]: This is indeed useful. I'd be tempted to combine this `docker run` with the one above which could then become:

```
$ results=/var/packaging/results
$ sudo mkdir -p "$results"
$ sudo docker run -v $PWD:/root/packaging -v /var/packaging/results:$results -it cc-osc bash
```

And just add a note the the command above to explain that the generated packages will be placed on the host system in the directory defined by `$results`.

**Comment by gorozco1** [comment]
@jodh-intel any update on this?

**Review by miguelinux — APPROVED** [review]
+1

Good idea for users that use CLR

**Pull request merged**